portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -60,16 +61,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("name");
     String message = request.getParameter("message");
     long timestamp = System.currentTimeMillis();
 
     Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("name", name);
     commentEntity.setProperty("message", message);
     commentEntity.setProperty("timestamp", timestamp);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     
-    if(message!=null && !message.equals("")){
+    if((message!=null && !message.equals("")) && (name!=null && !name.equals(""))){
```

Comment:
`"".equals(message)` will save you from nullability check. "" will never be null and the condition will be false if message is null.